src/convert.cpp

```diff
@@ -61,49 +73,60 @@ ddwaf_object* to_ddwaf_object_object(
   Napi::Object obj,
   int depth,
   bool lim,
-  bool ignoreToJSON
+  bool ignoreToJSON,
+  JsSet jsSet
 ) {
   if (!ignoreToJSON) {
     Napi::Value toJSON = obj.Get("toJSON");
     if (toJSON.IsFunction()) {
-      return to_ddwaf_object(object, env, toJSON.As<Napi::Function>().Call(obj, {}), depth, lim, true);
+      return to_ddwaf_object(object, env, toJSON.As<Napi::Function>().Call(obj, {}), depth, lim, true, jsSet);
     }
   }
 
+  if (jsSet.Has(obj)) {
```

Comment:
why not doing this check in `to_ddwaf_object` directly btw ?

Reply:
this check, and also the `jsSet.Add(obj)` calls.